Source/monster.cpp

```diff
@@ -3494,112 +3494,60 @@ void MAI_Cleaver(int i)
 
 void MAI_Round(int i, BOOL special)
 {
-	int v2;            // esi
-	MonsterStruct *v3; // esi
-	int v4;            // edx
-	int v5;            // ecx
-	int v6;            // edi
-	int v7;            // ebx
-	int v9;            // eax
-	//int v10; // ST04_4
-	int v12; // eax
-	//int v13; // ST04_4
-	int v15;     // edi
-	int v16;     // eax
-	int v17;     // ecx
-	BOOLEAN v18; // eax
-	//int v19; // eax
-	int v21; // eax
-	int v22; // eax
-	//int v23; // ST04_4
-	signed int v25; // ecx
-	int v26;        // eax
-	int v27;        // [esp+4h] [ebp-18h]
-	int v28;        // [esp+8h] [ebp-14h]
-	char *v29;      // [esp+8h] [ebp-14h]
-	int v30;        // [esp+Ch] [ebp-10h]
-	int md;         // [esp+10h] [ebp-Ch]
-	int v32;        // [esp+14h] [ebp-8h]
-	int arglist;    // [esp+18h] [ebp-4h]
+	MonsterStruct *Monst;
+	int fx, fy;
+	int mx, my, md;
+	int dist, v;
 
-	v2 = i;
-	v27 = special;
-	arglist = i;
 	if ((DWORD)i >= MAXMONSTERS)
 		app_fatal("MAI_Round: Invalid monster %d", i);
-	v3 = &monster[v2];
-	if (v3->_mmode == MM_STAND && v3->_msquelch) {
-		v4 = v3->_my;
-		v5 = v3->_mx;
-		v28 = (unsigned char)v3->_menemyy;
-		v6 = (unsigned char)v3->_menemyx;
-		v7 = v5 - v6;
-		v32 = v4 - v28;
-		md = GetDirection(v5, v4, v3->_lastx, v3->_lasty);
-		if (v3->_msquelch < 0xFFu) /* check sign */
-			MonstCheckDoors(arglist);
-		v30 = random(114, 100);
-		if ((abs(v7) >= 2 || abs(v32) >= 2) && v3->_msquelch == -1) {
-			v29 = &dTransVal[v6][v28];
-			if (dTransVal[v3->_mx][v3->_my] == *v29) {
-				if (_LOBYTE(v3->_mgoal) != MGOAL_MOVE) {
-					v9 = abs(v7);
-					//v11 = v10;
-					if (v9 < 4) {
-						v12 = abs(v32);
-						//v11 = v13;
-						if (v12 < 4)
-							goto LABEL_26;
-					}
-					if (random(115, 4))
-						goto LABEL_26;
-					if (_LOBYTE(v3->_mgoal) != MGOAL_MOVE) {
-						v3->_mgoalvar1 = 0;
-						v3->_mgoalvar2 = random(116, 2);
-					}
+	Monst = monster + i;
+	if (Monst->_mmode == MM_STAND && Monst->_msquelch) {
+		fy = Monst->_menemyy;
+		fx = Monst->_menemyx;
+		mx = Monst->_mx - fx;
+		my = Monst->_my - fy;
+		md = GetDirection(Monst->_mx, Monst->_my, Monst->_lastx, Monst->_lasty);
+		if ((BYTE)Monst->_msquelch < 255)

```

Comment:
Maybe we should use `UCHAR_MAX` for this sort of checks?

Reply:
If `_msquelch` was truly unsigned there's possibility that `255` was even separate constant specifically for squelch. But I agree that `UCHAR_MAX` is already a good first step.